src/httpUtils.ts

```diff
@@ -20,8 +20,14 @@ export function headersTokenAuth(token: string): Headers {
 }
 
 export function headersBasicAuth(username: string, password: string): Headers {
+	let auth;
+	if(username && password){
+		auth = Buffer.from(username + ":" + password).toString("base64");
+	} else {
+		auth = username;
```

Comment:
What kind of authentication is with just the username? Maybe we should create a dedicated auth flow?

Reply:
@davideicardi , that sounds good, I'll find a way were we can scope out NPM components.

Also, I'll modify this code with a best way to handle the scenario where we already have the base64 key from the consumer.